CorsixTH/Lua/date.lua

```diff
@@ -273,6 +273,19 @@ function Date_mt.__lt(one, other)
   return one._year < other._year
 end
 
+function Date_mt.__le(one, other)
+  if one._year == other._year then
+    if one._month == other._month then
+      if one._day == other._day then
+        return one._hour <= other._hour
+      end
+      return one._day <= other._day
+    end
+    return one._month <= other._month
+  end
+  return one._year <= other._year
+end
```

Comment:
```suggestion
function Date_mt.__le(one, other)
  if one._year ~= other._year then return one._year <= other._year end
  if one._month ~= other._month then return one._month <= other._month end
  if one._day ~= other._day then return one._day <= other._day end
  return one._hour <= other._hour
end
```
Writing the test the other way around avoids nesting.